ci/docker/linux-apt-python-3.dockerfile

```diff
@@ -23,6 +23,21 @@ COPY python/requirements-build.txt \
      /arrow/python/
 
 ENV ARROW_PYTHON_VENV /arrow-dev
+
+ARG ubuntu
+ARG python="3.12"
+COPY ci/scripts/install_python.sh /arrow/ci/scripts/
+# Ubuntu 20.04 uses python 3.8 which is unsuported for pyarrow
```

Comment:
Then we should stop testing Ubuntu 20.04 with PyArrow? What do you think @jorisvandenbossche ?

Reply:
I am going to bump the Python jobs to Ubuntu 22.04